lib/auth/init.go

```diff
@@ -502,18 +503,23 @@ func initCluster(ctx context.Context, cfg InitConfig, asrv *Server) error {
 	}
 	span.AddEvent("completed migration legacy resources")
 
-	span.AddEvent("creating preset roles")
 	// Create presets - convenience and example resources.
-	if err := createPresetRoles(ctx, asrv); err != nil {
-		return trace.Wrap(err)
-	}
-	span.AddEvent("completed creating preset roles")
+	isDashboard := !modules.GetModules().Features().Cloud && modules.GetModules().Features().RecoveryCodes
+	if !isDashboard {
+		span.AddEvent("creating preset roles")
+		if err := createPresetRoles(ctx, asrv); err != nil {
+			return trace.Wrap(err)
+		}
+		span.AddEvent("completed creating preset roles")
```

Comment:
The span will never complete if there's an error. Can we record the error case to the span as well?

Reply:
Attempted to address this here: [1054acc](https://github.com/gravitational/teleport/pull/37610/commits/1054acc35e06ef535174f1eb7923c57d20805686). 

For this I wasn't too sure if the way I addressed it is okay? I see in other cases that error is recorded but through a defer function instead.

Reply:
@rosstimothy can you advise?

Reply:
I reverted the changes here: [f88aaa0](https://github.com/gravitational/teleport/pull/37610/commits/f88aaa0386368f5e5a73a69c35ac86a458c7c334).